lib/Doctrine/DBAL/Driver/PDOPgSql/Driver.php

```diff
@@ -53,6 +53,14 @@ public function connect(array $params, $username = null, $password = null, array
                 $pdo->setAttribute(PDO::PGSQL_ATTR_DISABLE_PREPARES, true);
             }
 
+            /* defining client_encoding via SET NAMES to avoid inconsistent DSN support
+             * - the 'client_encoding' connection param only works with postgres >= 9.1
+             * - passing client_encoding via the 'options' param breaks pgbouncer support
+             */
+            if (isset($params['charset'])) {
+              $pdo->query('SET NAMES \''.$params['charset'].'\'');
```

Comment:
Sure, it works because you don't use client encoding in connection..
Il 02/apr/2015 20:42 "Kyle Renfrow" <notifications@github.com> ha scritto:

> In lib/Doctrine/DBAL/Driver/PDOPgSql/Driver.php
> <https://github.com/doctrine/dbal/pull/828#discussion_r27685139>:
>
> > @@ -53,6 +53,14 @@ public function connect(array $params, $username = null, $password = null, array
> >                  $pdo->setAttribute(PDO::PGSQL_ATTR_DISABLE_PREPARES, true);
> >              }
> >
> > +            /* defining client_encoding via SET NAMES to avoid inconsistent DSN support
> > +             * - the 'client_encoding' connection param only works with postgres >= 9.1
> > +             * - passing client_encoding via the 'options' param breaks pgbouncer support
> > +             */
> > +            if (isset($params['charset'])) {
> > +              $pdo->query('SET NAMES \''.$params['charset'].'\'');
>
> @riccardonar <https://github.com/riccardonar> you're throwing me for
> loops going back/forth with differing information.
>
> You said earlier that you tested my fork -- and that it worked. Is that
> true or not?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/doctrine/dbal/pull/828/files#r27685139>.
>